pkg/controllers/user/pipeline/controller/pipelineexecution/pipelineexecution.go

```diff
@@ -173,10 +183,22 @@ func (l *Lifecycle) Sync(obj *v3.PipelineExecution) (runtime.Object, error) {
 		if err := l.doCleanup(obj); err != nil {
 			return obj, err
 		}
+
+		shouldNotify, err := l.shouldNotify(obj)
+		if err != nil {
+			return obj, err
+		}
+		if shouldNotify {
+			newObj, err := v3.PipelineExecutionConditionNotified.Once(obj, func() (runtime.Object, error) {
+				return l.doNotify(obj)
+			})
+			return newObj.(*v3.PipelineExecution), err
+		}
 		//start a queueing execution if there is any
 		if err := l.startQueueingExecution(obj); err != nil {
```

Comment:
just to confirm that we don't need to call `startQueueingExecution` if `shouldNotify=true`

Reply:
we need to call that